diagrams/__init__.py

```diff
@@ -101,8 +101,9 @@ def __init__(
         :param edge_attr: Provide edge_attr dot config attributes.
         """
         self.name = name
-
-        if not filename:
+        if not name and not filename:
+          filename = "diagrams_file"
```

Comment:
Looks good to prevent an exception when there is no filename.

`diagrams_file` is good, but I think `diagrams_image` would be better because the output must be an image file.

Reply:
Thanks for the review, you are right it make more sense with image than file! 

I push the change.